libkbfs/data_types.go

```diff
@@ -223,9 +226,10 @@ func (r BlockRef) String() string {
 // NOTE: Don't add or modify anything in this struct without
 // considering how old clients will handle them.
 type BlockPointer struct {
-	ID      kbfsblock.ID `codec:"i"`
-	KeyGen  KeyGen       `codec:"k"` // if valid, which generation of the TLF{Writer,Reader}KeyBundle to use.
-	DataVer DataVer      `codec:"d"` // if valid, which version of the KBFS data structures is pointed to
+	ID       kbfsblock.ID `codec:"i"`
+	KeyGen   KeyGen       `codec:"k"`           // if valid, which generation of the TLF{Writer,Reader}KeyBundle to use.
+	DataVer  DataVer      `codec:"d"`           // if valid, which version of the KBFS data structures is pointed to
+	IsDirect bool         `codec:"s,omitempty"` // if true, the pointed-to block is definitely a leaf block
```

Comment:
Since we're bumping the data version, seems a shame to have `false` still mean indeterminate. How about making a three-state type, say `TargetBlockType` with `Indeterminate = 0`, `Direct = 1`, `Indirect = 2`? (Better names welcome)